src/libs/ReportUtils.ts

```diff
@@ -1140,8 +1140,16 @@ function isSystemChat(report: OnyxEntry<Report>): boolean {
  * Only returns true if this is our main 1:1 DM report with Concierge.
  */
 function isConciergeChatReport(report: OnyxInputOrEntry<Report>): boolean {
-    const participantAccountIDs = Object.keys(report?.participants ?? {}).filter((accountID) => Number(accountID) !== currentUserAccountID);
-    return participantAccountIDs.length === 1 && Number(participantAccountIDs[0]) === CONST.ACCOUNT_ID.CONCIERGE && !isThread(report);
+    if (!report?.participants || isThread(report)) {
+        return false;
+    }
+
+    const participantAccountIDs = new Set(Object.keys(report.participants));
+    if (participantAccountIDs.size !== 2) {
```

Comment:
@OlimpiaZurek  - the previous check was `participantAccountIDs.length === 1` - why are we checking `!= 2` to return false here? Shouldn't it be `>= 2`?

Reply:
The check  `if (participants.length !== 2)` ensures that there are exactly two participants in the report. The reason for this is that a concierge chat should involve the current user and the concierge only. If there are more or fewer participants, it cannot be considered a concierge chat.

It serves the same purpose as the combination of `Object.keys(report?.participants ?? {}).filter` and `participantAccountIDs.length === 1`

Reply:
maybe we can add a comment here to let people know what exactly do we mean by this check?

Reply:
oops too late haha